src/crimson/os/seastore/seastore.cc

```diff
@@ -2094,18 +2150,29 @@ SeaStore::Shard::_omap_set_values(
 {
   LOG_PREFIX(SeaStore::_omap_set_values);
   DEBUGT("{} {} keys", *ctx.transaction, *onode, aset.size());
-  return _omap_set_kvs(
-    onode,
-    onode->get_layout().omap_root,
-    *ctx.transaction,
-    std::move(aset)
-  ).si_then([onode, &ctx](auto root) {
-    if (root.must_update()) {
-      onode->update_omap_root(*ctx.transaction, root);
-    }
+  return seastar::do_with(
+    std::map<omap_type_t, std::map<std::string, ceph::bufferlist>>(),
+    [this, &onode, &ctx, aset=std::move(aset)](auto &sets) mutable {
+
+    BtreeOMapManager(*transaction_manager).get_root_types_with_sets(
+      sets, std::move(aset));
```

Comment:
If we choose the first approach, we’ll need to check the pattern with every set/get operation. The following code is required for this. Does this seem reasonable?

```
      if (str.size() == 31) {
        char* key = str.c_str();
        if (key[10] == '.' && key[31] == '0') {
          for (int i = 0; i < 10; i++) {
            if (!is_digit(key[i]) {
                break;
            }
          }
          for (int i = 11; i < 32; i++) {
            if (!is_digit(key[i]) {
                break;
            }
          }
          return true;
        }
      }
      return false;
```

Reply:
Hm, how about taking this chance to specialize `FuturizedStore` interfaces to handle pg logs in Crimson, instead of relying on seastore to check the omap key pattern implicitly?

This is also a necessary step towards https://github.com/ceph/ceph/pull/59213#issuecomment-2301132104, so the effort won't be wasted, even as an intermediate optimization.

@myoungwon are you comfortable with this direction?

Reply:
That sounds reasonable to me. I’ll repost the commits based on that direction.

Reply:
Can you please push the FuturizedStore change as a separate PR?

Reply:
@Matan-B Thanks for your comment. Will do.